src/databox/azext_databox/custom.py

```diff
@@ -0,0 +1,129 @@
+                       state_or_province,
+                       tags=None,
+                       mobile=None,
+                       street_address2=None,
+                       street_address3=None,
+                       company_name=None,
+                       storage_account_details=None,
+                       managed_disk_details=None):
+    body = {}
+    body['location'] = location  # str
+    body['tags'] = tags  # dictionary
+    body.setdefault('sku', {})['name'] = sku_name  # str
+    body.setdefault('details', {})['job_details_type'] = sku_name
+    body.setdefault('details', {}).setdefault('contact_details', {})['contact_name'] = contact_name  # str
+    body.setdefault('details', {}).setdefault('contact_details', {})['phone'] = phone  # str
+    body.setdefault('details', {}).setdefault('contact_details', {})['mobile'] = mobile  # str
+    body.setdefault('details', {}).setdefault('contact_details', {})['email_list'] = email_list
+    body.setdefault('details', {}).setdefault('shipping_address', {})['street_address1'] = street_address1  # str
+    body.setdefault('details', {}).setdefault('shipping_address', {})['street_address2'] = street_address2  # str
+    body.setdefault('details', {}).setdefault('shipping_address', {})['street_address3'] = street_address3  # str
+    body.setdefault('details', {}).setdefault('shipping_address', {})['city'] = city  # str
+    body.setdefault('details', {}).setdefault('shipping_address', {})['state_or_province'] = state_or_province  # str
+    body.setdefault('details', {}).setdefault('shipping_address', {})['country'] = country  # str
+    body.setdefault('details', {}).setdefault('shipping_address', {})['postal_code'] = postal_code  # str
+    body.setdefault('details', {}).setdefault('shipping_address', {})['company_name'] = company_name  # str
+
+    destination_account_details = []
+    if storage_account_details:
+        destination_account_details.append(storage_account_details)
+
+    if managed_disk_details:
+        destination_account_details.append(managed_disk_details)
+
+    body.setdefault('details', {})['destination_account_details'] = destination_account_details
+
+    return client.create(resource_group_name=resource_group_name, job_name=job_name, job_resource=body)
+
+
+def update_databox_job(client,
+                       resource_group_name,
+                       job_name,
+                       contact_name=None,
+                       phone=None,
+                       email_list=None,
+                       street_address1=None,
+                       postal_code=None,
+                       country=None,
+                       mobile=None,
+                       city=None,
+                       street_address2=None,
+                       street_address3=None,
+                       state_or_province=None,
+                       company_name=None):
+    job_resource = get_databox_job(client, resource_group_name, job_name)
```

Comment:
it's already a patch method. Why do we need to get it again?

Reply:
It seems the patch method in SDK is not correct. When just pass one property that need to be updated to SDK, it said some other properties cannot be None.

---
In reply to: [388072098](https://github.com/Azure/azure-cli-extensions/pull/1344#discussion_r388072098) [](ancestors = 388072098)